### PR TITLE
LPD-34378 - Email Address api

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-34378

### CODE REVIEW

File: modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java [hunk 1]
```diff
@@ -111,19 +119,27 @@ public void testUpgradeProcess() throws Exception {
 		throws PortalException;
 
 	protected void assertExternalReferenceCode(
-			String[] externalReferenceCodes, String tableName)
+			String[] externalReferenceCodes, String tableName,
+			boolean hasGroupIdColumn)
```
Comment (smottal) [inline modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java:123]: order params is wrong

File: modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java [hunk 2]
```diff
@@ -165,17 +181,23 @@ protected UpgradeProcess getUpgradeProcess() {
 	protected abstract Version getVersion();
 
 	protected void updateExternalReferenceCode(
-			String[] externalReferenceCodes, String tableName)
+			String[] externalReferenceCodes, String tableName,
+			boolean hasGroupIdColumn)
```
Comment (smottal) [inline modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java:185]: order

File: modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java [hunk 3]
```diff
@@ -76,6 +77,10 @@ public void setUp() throws Exception {
 			group.getGroupId());
 
 		ServiceContextThreadLocal.pushServiceContext(serviceContext);
+
+		try (Connection connection = dataSource.getConnection()) {
+			dbInspector = new DBInspector(connection);
+		}
```
Comment (smottal) [inline modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java:83]: why this here and not in the setUpClass?

File: modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java [hunk 4]
```diff
@@ -240,16 +264,18 @@ private List<IndexMetadata> _dropIndexes(String tableName)
 
 	private void _prepareDatabaseForUpgradeProcess(
 			ExternalReferenceCodeModel[] externalReferenceCodeModels,
-			String tableName)
+			String tableName, boolean hasGroupIdColumn)
```
Comment (smottal) [inline modules/apps/portal/portal-upgrade-test-util/src/main/java/com/liferay/portal/upgrade/test/util/BaseExternalReferenceCodeUpgradeProcessTestCase.java:267]: order

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:stop

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:report:17509685

**Pull request closed without merging**